packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L2DepositedERC20.sol

```diff
@@ -193,7 +193,27 @@ contract OVM_L2DepositedERC20 is iOVM_L2DepositedToken, OVM_CrossDomainEnabled,
         virtual
         onlyFromCrossDomainAccount(address(l1TokenGateway))
     {
-        // todo verify _l1Token matches
+        // Verify the deposited token on L1 matches the L2 deposited token representation here
+        // Otherwise immediately queue a withdrawal
+        if(_l1Token != address(l1Token)) {
+
+            bytes memory message = abi.encodeWithSelector(
+                iOVM_L1TokenGateway.finalizeWithdrawal.selector,
+                _l1Token,
+                address(this),
+                _from,
+                _to,
```

Comment:
I think that we need to make sure that the L1 `_from` address becomes the `_to` address here!
ie. we should return the funds to L1 address that sent them.

Reply:
As for who should be the L2 sender, I'm not sure, but my intuition is that it would be as if the L2 recipient is immediately sending the deposit back, which would result in just switching these two lines.